### PR TITLE
scala 2.12

### PR DESCRIPTION
**Pull request by zzeekk**
Maven profile and cleanup of dependencies for scala 2.12:
By default profile scala-2.11 is activated. Be careful if you set another profile like "fat-jar", default profiles are not activated and you need to give profile scala-2.11 manually.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -667,9 +653,9 @@
 		</dependency>
 
 		<dependency>
-			<groupId>com.github.jlmauduy</groupId>
+			<groupId>com.github.mutcianm</groupId>
```
[the remark this answers is not in the document]
Comment (zzeekk) [reply pom.xml:656]: There are several forks of ascii-graph. In features they dont differ a lot. But this one exists for scala 2.12.

**Comment by zzeekk** [comment]
> 1. I now get a Maven error because the artifactId isn't static:
>    'artifactId' contains an expression but should be a constant.
>    Isn't this what the maven rewrite plugin can be used for?
No, maven rewrite plugin replaces this in the packaged jar, but not during the original build...

> 2. When compiling with Scala 2.12 I get 6 warnings about matches not being exhaustive, i.e.
>    src/main/scala/io/smartdatalake/util/filetransfer/FileTransfer.scala:53
This is a new warning in Scala 2.12. The matches are already not exhaustive in 2.11... Shall we create an issue for this?

> 3. One deprecated warning that's new:
>    src/main/scala/io/smartdatalake/util/jms/JmsQueueConsumerFactory.scala:70: object JavaConversions in package collection is deprecated (since 2.12.0): use JavaConverters
>    7 warnings found
I can fix that one.

> 4. Maybe we could integrate the [Maven enforcer plugin](http://maven.apache.org/enforcer/enforcer-rules/index.html) so that you can't activate 2.11 and 2.12 at the same time. I didn't try it, but I can assume the outcome wouldn't be good.
I tried maven enforcer plugin but it can only ensure that at least one of these profiles are activated, but not that only one is activated (mutual exclusive). There seems to be no good solution for this in the maven community.

**Comment by pgruetter** [comment]
> No, maven rewrite plugin replaces this in the packaged jar, but not during the original build...

Do you see any other solution? We should really get rid of these build warnings.

> This is a new warning in Scala 2.12. The matches are already not exhaustive in 2.11... Shall we create an issue for this?

Yes, please create an issue, should be easy to fix.

> I tried maven enforcer plugin but it can only ensure that at least one of these profiles are activated, but not that only one is activated (mutual exclusive). There seems to be no good solution for this in the maven community.

You're right. I thought it's possible to handle mutual exlusions, but it only allows to check for "at least one". Let's leave it then.

**Comment by zzeekk** [comment]
I fixed all warnings (matches & deprecated).
For the maven properties in artifact name i know a solution. In my research i found the rewrite plugin, which fixes this at least for the final package... i would leave it like that, it works.

**Comment by pgruetter** [comment]
> I fixed all warnings (matches & deprecated).
> For the maven properties in artifact name i dont know a solution. In my research i found the rewrite plugin, which fixes this at least for the final package... i would leave it like that, it works.

OK, let's leave it for the moment. 
@danielspicar : Maybe you have an idea how we could solve this?

**Pull request merged**